db/Models.py

```diff
@@ -668,6 +668,13 @@ def find_all_by_organization_id_after_datetime(cls, organization_id, since):
     def find_all_by_device_auth_id(cls, dev_auth_data_id):
         return session.query(cls).filter(cls.device_auth_data_id == dev_auth_data_id).all()
 
+    @classmethod
+    def delete_keys(cls, device_auth_data_id, keys):
+        delete_query = cls.__table__.delete().\
+            where(cls.device_auth_data_id == device_auth_data_id).\
+            where(cls.app_key_hex.in_(keys))
+        session.execute(delete_query)
```

Comment:
The commit is not necessary?

Reply:
Nop, this is supposedly faster than fetching the objects, calling delete on them, and committing.